.github/workflows/CI.yml

```diff
@@ -29,104 +35,194 @@ jobs:
           toolchain: nightly
           override: true
           components: clippy
+
+      - name: Install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y libgeos-dev
       - run: cargo clippy -- -D warnings
       - run: cd sys && cargo clippy -- -D warnings
 
   check_static:
     name: Check static feature
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-20.04
     env:
       LD_LIBRARY_PATH: /usr/local/lib
+
     steps:
       - uses: actions/checkout@v2
         with:
-          submodules: 'true'
+          submodules: "true"
+
       - name: Install toolchain
         uses: actions-rs/toolchain@v1
         with:
           profile: minimal
           toolchain: stable
           override: true
-      - name: Install dependencies
-        run: |
-          sudo apt update
-          sudo apt install valgrind
-          sudo apt remove libgeos-dev -y
-      - name: Install geos
-        # First we install the 3.8 version of libgeos and then checkout the 3.8.1 version
-        run: |
-          git clone https://github.com/libgeos/geos
-          cd geos
-          git checkout 24650422b3982f17cc493fe92a70228f2ad624b4
-          ./autogen.sh
-          ./configure
-          make
-          sudo make install
-          sudo ldconfig
+
       - name: Build static geos crate
         run: cargo build --features static
 
   check:
-    name: Check ${{ matrix.toolchain }} / ${{ matrix.triple.target }}
-    runs-on: ubuntu-latest
+    name: Check ${{ matrix.toolchain }} / ${{ matrix.geos }} - ${{ matrix.triple.target }}
+    runs-on: ubuntu-20.04
     env:
       LD_LIBRARY_PATH: /usr/local/lib
+      GEOS_LIB_DIR: "/usr/lib/x86_64-linux-gnu"
+      GEOS_VERSION: ${{ matrix.geos }}
     strategy:
       fail-fast: false
       matrix:
         toolchain:
           - stable
           - nightly
+        geos:
+          - "3.6.5"
```

Comment:
`echo ${{matrix.geos}} | sed 's/\./_/g` should do the job. But I didn't see where you needed to have `X.Y.Z`.

Reply:
I ended up using the `include` statement to set these based on the GEOS version because the patch version of GEOS we use for testing (latest of each minor version) does not conform to the version feature, which is always patch `0`, e.g., `v3_6_0`.